### PR TITLE
[Personalization] fix overwrite editable blocks (https://github.com/pimcore/personalization-bundle/issues/61)

### LINKED ISSUE #61: [Personalization] Overwrite area block with target group does not work
This  behavior is only present in Pimcore 11 

## How to reproduce
1. On Pimcore demo website, open a CMS Document
2. Choose on target group
3. Overwrite an areablock
4. Remove it
5. Add a new areablock
6. Save.
7. Refresh the document. The new block is not present

## Why

In the post request to add a new areabrick, in field "realName" is missing prefix 'persona_XX-_'

You can see it on the screenshot below

image

This appears since Personnalization moved to his own bundle (https://github.com/pimcore/pimcore/commit/d505e46b8b9ff9505e2fb1f1bcb0616321a3b8a8) 

We should probably update this Pimcore Core file to fix this 
https://github.com/pimcore/pimcore/blob/d505e46b8b9ff9505e2fb1f1bcb0616321a3b8a8/models/Document/Editable.php#L524

### PR DESCRIPTION
**Pull request by ogmdev**
## Changes in this pull request  
Resolves https://github.com/pimcore/personalization-bundle/issues/61

### <samp>🤖 Generated by Copilot at ab51129</samp>

Fixed a bug in `Editable.php` that caused errors when working with targeting documents. Improved the compatibility of `buildEditableName` and `buildEditableRealName` functions with the `TargetingDocumentInterface`.

### <samp>🤖 Generated by Copilot at ab51129</samp>

> _`class_exists` fixed_
> _Targeting documents work_
> _Winter of errors_

### <samp>🤖 Generated by Copilot at ab51129</samp>

* Fix potential bug in `buildEditableName` and `buildEditableRealName` functions by using `interface_exists` instead of `class_exists` to check for `TargetingDocumentInterface` ([link](https://github.com/pimcore/pimcore/pull/16150/files?diff=unified&w=0#diff-629c3ab163921129218936956cf2db0761d23c958f2b554ad5e5225acc699cd8L560-R560), [link](https://github.com/pimcore/pimcore/pull/16150/files?diff=unified&w=0#diff-629c3ab163921129218936956cf2db0761d23c958f2b554ad5e5225acc699cd8L673-R673)) in file `models/Document/Editable.php`

### CODE REVIEW

File: models/Document/Editable.php [hunk 1]
```diff
@@ -557,7 +557,7 @@ public static function buildEditableName(string $type, string $name, Document $d
         // targeting prefix if configured on the document. hasBlocks() determines if
         // there are any parent blocks for the current element
         $targetGroupEditableName = null;
-        if ($document && class_exists(TargetingDocumentInterface::class) && $document instanceof TargetingDocumentInterface) {
+        if ($document && interface_exists(TargetingDocumentInterface::class) && $document instanceof TargetingDocumentInterface) {
```
Comment (jdreesen) [inline models/Document/Editable.php:560]: Do we actually need this check? Shouldn't this be enough?

```suggestion
        if ($document instanceof TargetingDocumentInterface) {
```

File: models/Document/Editable.php [hunk 2]
```diff
@@ -670,7 +670,7 @@ public static function buildEditableRealName(string $name, Document $document):
         // if element not nested inside a hierarchical element (e.g. block), add the
         // targeting prefix if configured on the document. hasBlocks() determines if
         // there are any parent blocks for the current element
-        if (class_exists(TargetingDocumentInterface::class) && $document instanceof TargetingDocumentInterface && !$blockState->hasBlocks()) {
+        if (interface_exists(TargetingDocumentInterface::class) && $document instanceof TargetingDocumentInterface && !$blockState->hasBlocks()) {
```
Comment (jdreesen) [inline models/Document/Editable.php:673]:
```suggestion
        if ($document instanceof TargetingDocumentInterface && !$blockState->hasBlocks()) {
```

**Comment by jdreesen** [comment]
Apart from that: it's really strange, that support for an external bundle is hard-coded like this in the core...

**Comment by dvesh3** [comment]
@ogmdev we need to add ignore [these](https://github.com/pimcore/pimcore/actions/runs/6731800477/job/18297212734?pr=16150) errors to [baseline](https://github.com/pimcore/pimcore/actions/runs/6731800477/job/18297212734?pr=16150) to make stan happy. could you please fix it? thanks!

**Comment by dvesh3** [comment]
Nevermind. I reverted the suggestions from jdreesen

**Comment by dvesh3** [comment]
> Apart from that: it's really strange, that support for an external bundle is hard-coded like this in the core...

@jdreesen yes, it is kind of a hard dependency at the moment since `Document\Page` is overridden int this bundle to support targeting and personalization which is then used by some other bundles e.g. CMF & ecommerce.

**Comment by dvesh3** [comment]
@ogmdev one last thing. please rebase the changes on 11.1 as it is a bug fix.

**Comment by dvesh3** [comment]
@ogmdev please check we still have unrelated commit here. thanks!

**Comment by dvesh3** [comment]
@ogmdev please check there are still unrelated commits in this PR. thanks!

**Comment by ogmdev** [comment]
> @ogmdev please check there are still unrelated commits in this PR. thanks!

Hello @dvesh3 
I don't really understand why after rebased branch upstream/11.1 to my current branch,  (with git-rebase) we still have unrelated commits.
Have you any ideas ?

Thanks

**Comment by dvesh3** [comment]
@ogmdev please create a new PR based on 11.1 and close this one. thanks!

**Comment by ogmdev** [comment]
@dvesh3 Thanks !
See new PR  https://github.com/pimcore/pimcore/pull/16207

**Pull request closed without merging**